homeassistant/components/azure_devops/sensor.py

```diff
@@ -1,126 +1,96 @@
 """Support for Azure DevOps sensors."""
 from __future__ import annotations
 
-from datetime import timedelta
-import logging
+from dataclasses import dataclass
+from typing import Any, Callable
 
 from aioazuredevops.builds import DevOpsBuild
-from aioazuredevops.client import DevOpsClient
-import aiohttp
 
 from homeassistant.components.azure_devops import AzureDevOpsDeviceEntity
-from homeassistant.components.azure_devops.const import (
-    CONF_ORG,
-    CONF_PROJECT,
-    DATA_AZURE_DEVOPS_CLIENT,
-    DATA_ORG,
-    DATA_PROJECT,
-    DOMAIN,
-)
-from homeassistant.components.sensor import SensorEntity
+from homeassistant.components.azure_devops.const import CONF_ORG, DOMAIN
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(seconds=300)
-PARALLEL_UPDATES = 4
+@dataclass
+class AzureDevOpsSensorEntityDescription(SensorEntityDescription):
+    """Class describing System Bridge sensor entities."""
 
-BUILDS_QUERY = "?queryOrder=queueTimeDescending&maxBuildsPerDefinition=1"
+    organization: str = ""
+    attrs: Callable[[DevOpsBuild], Any] = round
+    value: Callable[[DevOpsBuild], StateType] = round
```

Comment:
Maybe add `build_key: str | None = None` here as well, so you don't have to pass that as a separate value?